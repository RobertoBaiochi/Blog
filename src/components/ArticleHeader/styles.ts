import styled, { css } from 'styled-components';
import { Title as HeadingStyles } from '../Heading/styles';

export const Wrapper = styled.header`
  ${({ theme }) => css`
    padding-bottom: ${theme.spacing.xlarge};
    margin-bottom: ${theme.spacing.xlarge};
    border-bottom: 0.1rem solid ${theme.colors.mediumGray};

    ${HeadingStyles} {
      margin: 0;
      margin-bottom: ${theme.spacing.small};
    }
  `}
`;

export const Excerpt = styled.p`
  ${({ theme }) => css`
    margin: ${theme.spacing.medium} 0;
    font-size: ${theme.font.sizes.medium};
  `};
`;

export const Cover = styled.img`
  ${({ theme }) => css`
    display: block;
    margin-bottom: ${theme.spacing.medium};
  `};
`;
