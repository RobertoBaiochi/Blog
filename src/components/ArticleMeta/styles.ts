import styled, { css } from 'styled-components';

export const Wrapper = styled.div`
  ${() => css`
    width: 100%;
  `}
`;
