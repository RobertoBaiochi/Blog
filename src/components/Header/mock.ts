import { HeaderProps } from '.';
import { data } from '../../api/data.json';

const { setting } = data;
const {
  data: {
    id,
    attributes: { blogName, blogDescription },
  },
} = setting;

export default {
  id,
  blogName,
  blogDescription,
  logo: data.setting.data.attributes.logo.data,
} as unknown as HeaderProps;
