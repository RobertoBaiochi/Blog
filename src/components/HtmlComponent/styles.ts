import styled, { css } from 'styled-components';

export const Container = styled.p`
  ${({ theme }) => css`
    font-size: ${theme.font.sizes.medium};

    p {
      margin: ${theme.spacing.large} 0;
    }

    a,
    a:visited,
    a:link {
      color: ${theme.colors.secondary};
      text-decoration: none;
      transition: all 300ms ease-in-out;
    }

    a:hover {
      filter: brightness(50%);
    }

    pre {
      width: 100%;
      background: ${theme.colors.primary};
      padding: ${theme.spacing.medium};
      font-family: monospace;
      color: ${theme.colors.white};
      margin: ${theme.spacing.xlarge} 0;
      overflow-x: auto;
      font-size: ${theme.font.sizes.small};
    }

    .image {
      background: ${theme.colors.mediumGray};
      line-height: 0;
      margin: ${theme.spacing.xlarge};
    }

    .image figcaption {
      font-size: ${theme.font.sizes.small};
      padding: ${theme.spacing.small};
      text-align: center;
      line-height: 1.3;
    }

    .image-style-side {
      max-width: 50%;
      float: right;
      margin: ${theme.spacing.medium};
    }

    hr {
      border: none;
      border-bottom: 0.1rem solid ${theme.colors.mediumGray};
    }

    ul, ol {
      margin: ${theme.spacing.xlarge};
    }

    @media ${theme.media.lteMedium} {
        font-size: 2rem;

        .image-style-side {
          max-width: 100%;
          float: none;
          margin: 0;
        }
      }

  `};
`;
