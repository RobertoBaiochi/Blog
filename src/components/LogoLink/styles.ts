import styled, { css } from 'styled-components';

export const ContainerLink = styled.a`
  ${() => css`
    display: flex;
    align-items: center;
    text-decoration: none;
    color: inherit;
  `};
`;

export const Container = styled.div`
  ${() => css`
    display: flex;
    align-items: center;
    text-decoration: none;
    color: inherit;
  `};
`;

export const ImageLogo = styled.img`
  ${() => css`
    width: 15rem;
    height: 15rem;
    border-radius: 50%;
  `};
`;
