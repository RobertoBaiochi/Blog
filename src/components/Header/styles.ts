import styled, { css } from 'styled-components';

import { Title as HeadingHeader } from '../Heading/styles';

export const HeaderContainer = styled.header`
  ${({ theme }) => css`
    width: 100%;
    max-width: ${theme.sizes.maxWidth};
    margin-inline: auto;
    display: flex;
    align-items: center;
    justify-content: center;
    padding-bottom: ${theme.spacing.huge};
    color: ${theme.colors.darkText};
    font-size: ${theme.font.sizes.small};

    ${HeadingHeader} {
      margin: 0 0 calc(${theme.spacing.small} - 3rem);
    }
  `}
`;

export const TextContainer = styled.div`
  ${({ theme }) => css`
    max-width: 48rem;
    height: 100%;
    display: flex;
    flex-flow: column wrap;
    margin-left: ${theme.spacing.large};
  `};
`;
