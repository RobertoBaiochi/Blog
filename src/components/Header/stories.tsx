import { Meta, StoryFn } from '@storybook/react';
import Header, { HeaderProps } from '.';

import mock from './mock';

export default {
  title: 'Header',
  component: Header,
  args: mock,
} as Meta<HeaderProps>;

export const Template: StoryFn<HeaderProps> = (args) => {
  return (
    <div>
      <Header {...args} />
    </div>
  );
};

/*
  fazendo a alteração para a forma mais robusta de servir a API
  os Controls são alterados para objetos que necessitam de alguma forma
  que deve ser pesquisida para manter a possibilidade de fazer alterações
*/
