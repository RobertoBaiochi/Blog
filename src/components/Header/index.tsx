import { LogoStrapi } from '@/shared-typed/logo';
import Heading from '../Heading';
import LogoLink from '../LogoLink';
import { HeaderContainer, TextContainer } from './styles';

export type HeaderProps = {
  blogName: string;
  blogDescription: string;
  logo: LogoStrapi;
} & {
  showText?: boolean;
};

const Header = ({
  blogName,
  blogDescription,
  logo,
  showText = true,
}: HeaderProps) => {
  const {
    attributes: { url },
  } = logo.data;

  return (
    <HeaderContainer>
      <LogoLink link="/" srcImg={url} text={blogName} />

      {showText && (
        <TextContainer>
          <Heading size="small" as="h2" colorDark={true}>
            {blogName}
          </Heading>
          <p>{blogDescription}</p>
        </TextContainer>
      )}
    </HeaderContainer>
  );
};

export default Header;
