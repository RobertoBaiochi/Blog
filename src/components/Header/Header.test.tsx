import { screen } from '@testing-library/react';
import { renderTheme } from '../../styles/render-theme';
import Header from '.';
import '@testing-library/jest-dom';

import props from './mock';

describe('<Header />', () => {
  it('should render an image, a heading and text', () => {
    renderTheme(<Header {...props} showText={undefined} />);

    const altImage = screen.getByAltText(props.blogName);
    const descriptionBlog = screen.getAllByText(props.blogDescription)[0];
    const blogTitle = screen.getByRole('heading', {
      name: props.blogName,
    });
    const image = screen.getByRole('img', { name: props.blogName });

    expect(altImage).toBeInTheDocument();
    expect(descriptionBlog).toBeInTheDocument();
    expect(blogTitle).toBeInTheDocument();
    expect(image).toHaveAttribute('src', props.logo.attributes.url);
  });

  it('should render image only', () => {
    renderTheme(<Header {...props} showText={false} />);

    const image = screen.getByRole('img', { name: props.blogName });
    const blogTitle = screen.queryByRole('heading', {
      name: props.blogName,
    });
    const descriptionBlog = screen.queryByText(props.blogDescription);

    expect(image).toHaveAttribute('src', props.logo.attributes.url);
    expect(blogTitle).not.toBeInTheDocument();
    expect(descriptionBlog).not.toBeInTheDocument();
  });

  it('should match snapshot', () => {
    const { container } = renderTheme(
      <Header {...props} showText={undefined} />,
    );

    expect(container).toMatchSnapshot();
  });
});
