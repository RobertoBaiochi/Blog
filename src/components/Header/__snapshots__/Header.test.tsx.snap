// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> should match snapshot 1`] = `
<div>
  <header
    class="sc-3dbff527-0 gppyNM"
  >
    <a
      href="/"
      target="_self"
    >
      <div
        class="sc-d76a3d34-1 jbTDAv"
      >
        <img
          alt="Console.Blog"
          class="sc-d76a3d34-2 gsyyHr"
          src="https://res.cloudinary.com/dfmbf9sae/image/upload/v1683608645/carta_410436ad20.jpg"
        />
      </div>
    </a>
    <div
      class="sc-3dbff527-1 iryleV"
    >
      <h2
        class="sc-b5aaf39e-0 kQHlLj"
      >
        Console.Blog
      </h2>
      <p>
        Debuggando passo a passo do conhecimento
      </p>
    </div>
  </header>
</div>
`;
